src/components/editor/metaform-editor-right-drawer-visibility.tsx

```diff
@@ -0,0 +1,338 @@
+      setDisabledValue(section !== undefined);
+      setRequiredConditionField(section ? section.field! : "");
+      setRequiredConditionEqualsValue(section ? section.equals! : "");
+      setRequiredConditionInfoField(!section);
+      setSelectedComponent(pendingForm.sections![sectionIndex].title);
+    }
+
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      const field = pendingForm.sections![sectionIndex].fields![fieldIndex].visibleIf;
+      setMetaFormFieldSection(pendingForm.sections![sectionIndex].fields![fieldIndex]);
+      setSelectedComponent(pendingForm.sections![sectionIndex].fields![fieldIndex].name);
+      setDisabledValue(field !== undefined);
+      setRequiredConditionField(field ? field.field! : "");
+      setRequiredConditionEqualsValue(field ? field.equals! : "");
+      setRequiredConditionInfoField(!field);
+    }
+  };
+  
+  useEffect(() => {
+    setVisiblityComponentValues();
+  }, [fieldIndex, sectionIndex]);
+
+  /**
+   * Updates metaform field
+   *
+   * @param newMetaformField new metaform field
+   */
+  const emptyFormFieldVisibleIf = (newMetaformField: MetaformField) => {
+    if (sectionIndex === undefined || fieldIndex === undefined) {
+      return;
+    }
+    const updatedForm = produce(pendingForm, draftForm => {
+      draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, newMetaformField);
+    });
+    setPendingForm(updatedForm);
+  };
+
+  /**
+   * Updates metaform field or section visibleIf parameters
+   *
+   * @param newMetaformUpdateFieldOrSection metaform field or section
+   * @param value event condition equals value (yes / no)
+   */
+  const updateFormFieldOrSectionVisiblityValues = (newMetaformUpdateFieldOrSection: MetaformField | MetaformSection, value: string) => {
+    // TODO const amountOfRadios = pendingForm.sections!.map(x => x.fields?.filter(y => y.type === MetaformFieldType.Radio)).flat().length;
+    setRequiredConditionEqualsValue(value);
+    setRequiredConditionInfoField(false);
+    if (sectionIndex === undefined) {
+      return;
+    }
+    if (sectionIndex !== undefined && fieldIndex === undefined) {
+      const updatedForm = produce(pendingForm, draftForm => {
+        draftForm.sections?.splice(sectionIndex, 1, newMetaformUpdateFieldOrSection as MetaformSection);
+      });
+      setPendingForm(updatedForm);
+    }
+    if (sectionIndex !== undefined && fieldIndex !== undefined) {
+      const updatedForm = produce(pendingForm, draftForm => {
+        draftForm.sections?.[sectionIndex]?.fields?.splice(fieldIndex, 1, newMetaformUpdateFieldOrSection as MetaformField);
+      });
+      setPendingForm(updatedForm);
+    }
+  };
+
+  /**
+   * Updates metaform section
+   *
+   * @param newMetaformSection new metaform section
+   */
+  const emptyFormSectionVisibleIf = (newMetaformSection: MetaformSection) => {
+    if (sectionIndex === undefined) {
+      return;
+    }
+
+    const section = pendingForm.sections?.[sectionIndex];
```

Comment:
```suggestion
    const section = pendingForm.sections?[sectionIndex];
```
_sectionIndex_ is of type number and it should be used as an array index and not as a property of its own.